### ISSUE TITLE
Adding existing towers and other buildings that are already placed on the map to the Building Helper grid.

### ISSUE THREAD

**Issue by Myll** [Myll/Dota-2-Building-Helper#5]
**Taken from a question sent to my email:**

I'm trying to add my existing towers and other buildings that are already placed on the map to the Building Helper grid.

I'm doing this via the following code:

    local prevEnt
    for i=1, #ALL_LANE_BUILDINGS do 
        local buildingName = ALL_LANE_BUILDINGS[i]
        buildingNameEnt = Entities:FindByName(prevEnt, buildingName)
        -- loop through and get the absOrigin of the buidlings
        local buildingAbsOrigin = buildingNameEnt:GetAbsOrigin()
        local buildingMaxBounds = buildingNameEnt:GetBoundingMaxs()
        --we only need one side as per the docs of building helper
        local x = buildingMaxBounds.x
        --local BHSize = x/64
        local BHSize = 2.5
        local BH = BuildingHelper:AddBuildingToGrid(buildingAbsOrigin, BHSize, nil)
        if BH == -1 then
            print ("HWDEBUG: Building Helper failed to add something to the grid. Check the next line.")
            print (unitName)
        end
    end

Nothing fails to add to the grid.

In order to pass 'nil' as the hBuilder value I add to modify a line of code in buildinghelper.lua. Line 248 in version 0.5.

                else
                    print ("DEBUGGING")
                    --hBuilder.bNeedsToJump=true
                end

This prints for every building in my for loop. This doesn't particularly affect the grid though. I implemented the 'buildings' command as per the bottom of the buildinghelper.lua file. I've attached a screen shot of the grid (the before picture) that shows the grid that is generated by the above block of code.

The after picture depicts my peon still building on top of the towers' existing grid. I also built a tower in the after picture with the peon off to the right. I cannot built another building on top of the grid that is created by this tower that is created by the peon. 

Before
![](http://i.imgur.com/i3VjS2Z.jpg)

After
![](http://i.imgur.com/KHZbwJM.jpg)

**Comment by bbqnoobstew** [comment]
I figured out the problem. The offset of the grid from the towers is a result of passing a size in to BuildingHelper:AddBuildingToGrid(buildingAbsOrigin, 2, nil) that was a value BuildingHelper didn't like. I had used values of 2.5 and 2.37 mistakenly. I looked through the code and found that the size parameter of AddBuildingToGrid() had to be size%2=0. Once I used sizes that respected that formula I as good to go.
